internal/dot/graph.go

```diff
@@ -252,6 +274,59 @@ func (dg *Graph) addToGroup(r *Result, id CtorID) {
 	group.Results = append(group.Results, r)
 }
 
+// Prune removes elements from the graph that do not have failed results.
+// Removing elements that do not have failing results makes the graph easier to debug, since non-failing
+// nodes and edges can clutter the graph and don't help the user debug.
+func (dg *Graph) Prune() {
+	dg.pruneCtors()
+	dg.pruneGroups()
+}
+
+// Remove constructors from the graph that do not have failing Results.
+func (dg *Graph) pruneCtors() {
+	var prunedCtors []*Ctor
```

Comment:
nit: given the number of variables in context, you can use shorter variable
names here. Go prefers short variable names.

https://github.com/golang/go/wiki/CodeReviewComments#variable-names

Reply:
I changed all variables of a similar purpose to pruned.